consumer_test.go

```diff
@@ -1,38 +1,37 @@
 package sarama
 
 import (
-	"log"
-	"os"
-	"os/signal"
 	"sync"
 	"testing"
 	"time"
 )
 
-func TestConsumerOffsetManual(t *testing.T) {
-	seedBroker := newMockBroker(t, 1)
-	leader := newMockBroker(t, 2)
-
-	metadataResponse := new(MetadataResponse)
-	metadataResponse.AddBroker(leader.Addr(), leader.BrokerID())
-	metadataResponse.AddTopicPartition("my_topic", 0, leader.BrokerID(), nil, nil, ErrNoError)
-	seedBroker.Returns(metadataResponse)
+var testMsg = StringEncoder("Foo")
 
-	offsetResponseNewest := new(OffsetResponse)
-	offsetResponseNewest.AddTopicPartition("my_topic", 0, 2345)
-	leader.Returns(offsetResponseNewest)
-
-	offsetResponseOldest := new(OffsetResponse)
-	offsetResponseOldest.AddTopicPartition("my_topic", 0, 0)
-	leader.Returns(offsetResponseOldest)
+// If a particular offset is provided then messages are consumed starting from
+// that offset.
+func TestConsumerOffsetManual(t *testing.T) {
+	// Given
+	broker0 := newMockBroker(t, 0)
+	defer broker0.Close()
```

Comment:
we haven't been using defers in tests because if one of them deadlocks (which is a common "found a bug!" indication) then it's really hard to track down